packages/langchain/package.json

```diff
@@ -0,0 +1,48 @@
+{
+  "name": "@sap-ai-sdk/langchain",
+  "version": "0.0.0",
+  "description": "",
+  "license": "Apache-2.0",
+  "keywords": [
+    "sap-ai-sdk",
+    "gen-ai-hub",
+    "orchestration",
+    "llm-access"
+  ],
+  "type": "module",
+  "main": "./dist/index.js",
+  "types": "./dist/index.d.ts",
+  "files": [
+    "dist/**/*.js",
+    "dist/**/*.js.map",
+    "dist/**/*.d.ts",
+    "dist/**/*.d.ts.map",
+    "internal.js",
+    "internal.d.ts"
+  ],
+  "scripts": {
+    "compile": "tsc",
+    "compile:cjs": "tsc -p tsconfig.cjs.json",
+    "test": "NODE_OPTIONS=--experimental-vm-modules jest",
+    "lint": "eslint \"**/*.ts\" && prettier . --config ../../.prettierrc --ignore-path ../../.prettierignore -c",
+    "lint:fix": "eslint \"**/*.ts\" --fix && prettier . --config ../../.prettierrc --ignore-path ../../.prettierignore -w --log-level error",
+    "check:public-api": "node --loader ts-node/esm ../../scripts/check-public-api-cli.ts"
+  },
+  "dependencies": {
+    "@sap-ai-sdk/ai-api": "workspace:^",
+    "@sap-ai-sdk/foundation-models": "workspace:^",
+    "@langchain/core": "^0.2.30",
+    "zod-to-json-schema": "^3.23.2"
+  },
+  "peerDependencies": {
+    "@langchain/openai": "^0.2.8"
+  },
+  "peerDependenciesMeta": {
+    "@langchain/openai": {
+      "optional": true
```

Comment:
[q] while I'm generally in favor of the optional approach, I'm wondering if it's worth it. Currently, the [langchain openai package](https://www.npmjs.com/package/@langchain/openai?activeTab=dependencies) brings only 4 dependencies. In addtion, our `foundation-models` package brings _all_ foundation models anyway. How much stuff would `@langchain/openai` add?

Reply:
Well `langchain/openai` would always be on top of our types, as there is no overlap.
As the number of vendors grows this package would also grow, therefore I would go for this approach regardless.
I suppose it would not be that easy for foundation models, since an alternative approach would require us to create separate packages for all vendors.

If many support issues start popping up because users do not properly install the required packages, we can reconsider of course.

Reply:
> Well langchain/openai would always be on top of our types, as there is no overlap.

Hmm not sure I understand that correctly. My understanding is that, if the user doesn't import it, running `new OpenAiChatClient({ modelName: 'gpt-35-turbo' });` will fail. And further, that `@sap-ai-sdk/langchain` already depends on `foundation-models`, which will bring all vendors anyway. 

Not sure what you mean with overlap, but essentially not including `langchain/openai` is worth it if and only if it would bring a bunch of unnecessary packages, right?

Reply:
Yes or rather if the package size increases by (arguably) too much by including every single vendor.
From a quick glimpse on npm the average package size seems to be around 200-400kb * vendors (assuming no overlap, although there might be some), I would consider especially for users who are wary of big package sizes (e.g. cap) smaller packages are certainly the favorable outcome.

Reply:
AFAI can see, [langchain/openai](https://www.npmjs.com/package/@langchain/openai?activeTab=dependencies) only brings one additional dependency: [openai](https://www.npmjs.com/package/openai). The other ones overlap with [langchain/core](https://www.npmjs.com/package/@langchain/core?activeTab=dependencies). 

However, `openai` itself is pretty large with 2.5MB, so probably still better to have it optional then.